website/src/components/Messages/MessageTableEntry.tsx

```diff
@@ -155,16 +158,47 @@ const MessageActions = ({
   onReport: () => void;
   message: Message;
 }) => {
+  const toast = useToast();
   const { t } = useTranslation(["message", "common"]);
-  const { trigger } = useSWRMutation(`/api/admin/delete_message/${message.id}`, del);
-  const { data } = useSession() || {};
-  const role = data?.user?.role;
+  const id = message.id;
+  const displayId = id.slice(0, CHAR_COUNT) + "..." + id.slice(-CHAR_COUNT);
+  const { trigger: deleteMessage } = useSWRMutation(`/api/admin/delete_message/${message.id}`, del);
+
+  const { trigger: stopTree } = useSWRMutation(`/api/admin/stop_tree/${message.id}`, put, {
+    onSuccess: () => {
+      toast({
+        title: t("common:success"),
+        description: t("tree_stopped", { id: displayId }),
+        status: "success",
+        duration: 5000,
+        isClosable: true,
+      });
+    },
+  });
+
+  const { data: session } = useSession();
+  const isAdmin = session?.user?.role === "admin";
 
   const handleDelete = async () => {
-    await trigger();
+    await deleteMessage();
     mutate((key) => typeof key === "string" && key.startsWith("/api/messages"), undefined, { revalidate: true });
   };
 
+  const handleStop = () => {
+    stopTree();
+  };
+
+  const handleCopyId = () => {
+    navigator.clipboard.writeText(message.id);
+    toast({
+      title: t("common:copied"),
+      description: displayId,
+      status: "info",
+      duration: 5000,
+      isClosable: true,
+    });
+  };
```

Comment:
CharkaUI has a useClipboard hook, we should use it here

Reply:
If you look here https://github.com/chakra-ui/chakra-ui/issues/6759 there is still some confusion regarding this hook, so I think it's simpler and cleaner to go with native solution for now.

Reply:
I'm not quite sure what the confusion is with useClipboard.  That mentioned issue is closed and in the [hook docs](https://chakra-ui.com/docs/hooks/use-clipboard) they show how to use `setValue`.  Can we use that following their example?

Reply:
Also, why do we need to copy the id to the clipboard?

Reply:
I welcome you to try that. setValue does not work as expected and you need to use an effect to make it work because we don't have an on change function. Read the last comment in the tracked issue. I added this feature because there's no easy way to know what message you are deleting or halting on the UI.